### PR TITLE
Set SSG Test Names

### PR DESCRIPTION
**Pull request by jasonwalker80**
- a command to set test names on all SSG results (excluding aligner index, etc.)
- wait to abandon builds until we have a complete list and test names are set

### CODE REVIEW

File: lib/perl/Genome/Model/SingleSampleGenotype/Command/SetTestName.pm [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            is      => "Genome::Model::Build::SingleSampleGenotype",
+            is_many => 1,
+        },
+        new_test_name => {
+            is  => 'String',
+            doc => 'The new test name string to set on all results'
+        },
+        abandon_builds => {
+            is => 'Boolean',
+            doc => 'attempt to abando all of the builds used by these results',
```
Comment (tmooney) [inline lib/perl/Genome/Model/SingleSampleGenotype/Command/SetTestName.pm]: `abandon` is short its final `n`.

File: lib/perl/Genome/Model/SingleSampleGenotype/Command/SetTestName.pm [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+};
+
+sub execute {
+    my $self = shift;
+
+    my $result_classes_to_set_test_name = $self->_result_classes_to_set_test_name;
+    my @results_to_set_test_name;
+    for my $build ( $self->builds ) {
+        my @results = $build->results;
+        push @results_to_set_test_name, grep { $result_classes_to_set_test_name->{$_->class} } @results;
```
Comment (tmooney) [inline lib/perl/Genome/Model/SingleSampleGenotype/Command/SetTestName.pm]: The database/UR can do this restriction for us, e.g.:
```perl
$build->results(subclass_name => [
  'Genome::Qc::Result',
  'Genome::InstrumentData::AlignmentResult::Speedseq',
  'Genome::InstrumentData::AlignmentResult::Merged::Speedseq',
  'Genome::Model::SingleSampleGenotype::Result::HaplotypeCaller'
]);
```

**Comment by jasonwalker80** [comment]
@tmooney will the latest commit work for "whitelisting" result classes?

File: lib/perl/Genome/SoftwareResult/Command/SetTestName.pm [hunk 3]
```diff
@@ -46,21 +49,29 @@ sub execute {
...
         }
     }
+    if ($self->abandon_builds) {
+        my @ids = keys %builds;
+        my @builds = values %builds;
+        $self->status_message('Abandoning builds: '. join(',',@ids));
+        my $abandon_cmd = Genome::Model::Build::Command::Abandon->create(
+            builds => \@builds,
+            header_text => 'Build Abandoned - SR Test Name',
+            body_text => 'Software result '. $software_result->id .' has new test name '. $self->new_test_name,
```
Comment (tmooney) [inline lib/perl/Genome/SoftwareResult/Command/SetTestName.pm]: With this pulled out of the loop `$software_result` is no longer available.

**Comment by jasonwalker80** [comment]
Yes, the intent is to eventually define `_disk_usage_result_subclass_names` for all model types and `disk_usage_allocations` could then rely on that instead. I have a feeling there will be issues with backward compatibility for old Workflow/Flow builds and such, but at some point we can ignore that.

**Pull request merged**